src/Codeception/Util/Uri.php

```diff
@@ -20,6 +20,10 @@ public static function mergeUrls($baseUri, $uri)
     {
         $base = new Psr7Uri($baseUri);
         $parts = parse_url($uri);
+
+        //If the relative URL does not parse, attempt to parse the entire URL.
+        //PHP Known bug ( https://bugs.php.net/bug.php?id=70942 )
+        $parts = ($parts === false)?parse_url($baseUri):$parts;
```

Comment:
1. Please change ternary operator to if statement:
```php
if ($parts === false) {
  $parts = parse_url($baseUri);
}
```

2. It looks like you forgot to concatenate $baseUri and $uri.